lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -0,0 +1,359 @@
+#[macro_use]
+extern crate lazy_static;
+
+use libc::{fd_set, select, FD_SET};
+use rdp::core::client::{Connector, RdpClient};
+use rdp::core::event::*;
+use rdp::model::error::*;
+use std::collections::HashMap;
+use std::convert::TryFrom;
+use std::mem;
+use std::net::TcpStream;
+use std::os::unix::io::AsRawFd;
+use std::ptr;
+use std::sync::{Arc, Mutex};
+
+struct Client {
+    rdp_client: RdpClient<TcpStream>,
+    tcp_fd: usize,
+}
+type SyncRdpClient = Arc<Mutex<Client>>;
+
+// Rust-side registry of clients, to allow Go to reference a specific client in calls after
+// connect_rdp.
+lazy_static! {
+    static ref RDP_CLIENTS: Arc<Mutex<HashMap<i64, SyncRdpClient>>> =
+        Arc::new(Mutex::new(HashMap::new()));
+}
+
+fn register_client(client_ref: i64, client: Client) {
+    RDP_CLIENTS
+        .lock()
+        .unwrap()
+        .insert(client_ref, Arc::new(Mutex::new(client)));
+}
+
+fn unregister_client(client_ref: &i64) {
```

Comment:
Primitive integers like `i64` should never be taken by immutable reference. This is because the reference is just as big and thus you win no performance and LLVM can't optimize this unless it decides to inline thus function. Generally types which implement the `Copy` trait from `std` are small and designed to be copied instead of immutably referenced.